### PR TITLE
Moving to JavaScript random

### PR DESCRIPTION
**Pull request by pelikhan**
- [x] Math.random returns double between 0 and 1 (exclusive)
- [x] Math.randomRange(min, max) follows scratch semantics
- [x] remove special blocks decompiler support for random
- [x] docs

### CODE REVIEW

File: pxtsim/libgeneric.ts [hunk 1]
```diff
@@ -196,13 +196,20 @@ namespace pxsim {
...
+            if (min == max) return min;
+            if (min > max) {
+                let t = min;
+                min = max;
+                max = t;
+            }
+            if (Math.floor(min) == min && Math.floor(max) == max)
+                return min + Math.round(Math.random() * (max - min));
+            else
+                return min + Math.random() * (max - min);
```
Comment (pelikhan) [inline pxtsim/libgeneric.ts]: max cannot be hit in this case

Comment (mmoskal) [inline pxtsim/libgeneric.ts]: but you can get arbitrarly close to it

File: common-docs/reference/math/random-range.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# Random Range
+
+Returns a pseudo-random number in the range ``[min, max];`` that is, from ``0`` (inclusive) up to including ``1`` (inclusive), which you can then scale to your desired range.
```
Comment (mmoskal) [inline common-docs/reference/math/random-range.md]: We need to decide if this is a floating point number or integer. I think switching based on the bounds being integer isn't the best design. Maybe rename it randomInt ?

Comment (pelikhan) [inline common-docs/reference/math/random-range.md]: I was trying to map scratch: https://wiki.scratch.mit.edu/wiki/Pick_Random_()_to_()_(block)

Comment (pelikhan) [inline common-docs/reference/math/random-range.md]: I agree that integers would probably be better in our case.

**Pull request merged**